templates.ts

```diff
@@ -75,41 +49,134 @@ export function generateRendererFile(
 
     const indentStr = indent !== undefined ? indent : 4;
 
-    const optionsSet = `options = { defaultFormatOptions | newLines = ${newLinesStr}, indent = ${indentStr} }`;
+    return `options = { defaultFormatOptions | newLines = ${newLinesStr}, indent = ${indentStr} }`;
+}
+
+export interface ViewFunctionConfig {
+    viewFunction: string;
+    viewHash: string;
+    model?: any;
+    decoder?: string;
+    indent?: number;
+    newLines?: boolean;
+}
+
+function renderCommandWithDecoder(viewHash: string, viewFunction: string, decoderName: string, optionsSet: string) {
+    return `
+render${viewHash} : Json.Value -> String
+render${viewHash} values =
+    let
+        ${optionsSet}
+    in
+        case Json.decodeValue ${decoderName} values of
+            Err err ->
+                "I could not decode the argument for ${viewFunction}:" ++ err
+
+            Ok model ->
+                (decode options) <| ${viewFunction} model
+        `;
+        }
+
+function renderCommandWithoutDecoder(viewHash: string, viewFunction: string, optionsSet: string) {
+    return `
+render${viewHash} : Json.Value -> String
+render${viewHash} _ =
+    let
+        ${optionsSet}
+    in
+        (decode options) <| ${viewFunction}
+        `;
+}
+
+function generateBody(config: ViewFunctionConfig): string {
+    const optionsSet = generateOptionsSet(config.newLines, config.indent);
+    if (config.decoder) {
+        console.log("withDecoder ->", config.decoder, config.viewFunction);

```

Comment:
is this for debugging? if so, make it an optional thing. i used `HIDE_WARNINGS` for this on master